openmdao/vectors/petsc_vector.py

```diff
@@ -123,41 +143,29 @@ def get_norm(self):
         system = self._system
         comm = system.comm
         if comm.size > 1:
-            dup_slice = self._dup_slice
-            if dup_slice is None:
-
-                # Here, we find the indices that are not locally owned so that we can
-                # temporarilly zero them out for the norm calculation.
-                dup_slice = []
-                abs2meta = system._var_allprocs_abs2meta
-                for name, idx_slice in iteritems(self.get_slice_dict()):
-                    owning_rank = system._owning_rank[name]
-                    if not abs2meta[name]['distributed'] and owning_rank != system.comm.rank:
-                        dup_slice.extend(range(idx_slice.start, idx_slice.stop))
-
-                self._dup_slice = dup_slice
+            dup_inds = self._get_dup_inds()
 
             if self._ncol == 1:
-                data_cache = self._data.copy()
+                data_cache = self._data.copy().real
```

Comment:
Why are we only caching and restoring the real part? Doesn't this remove the imaginary part when we are under complex step?

Reply:
I don't think it actually matters what we assign the self._petsc.array to, because we always reassign it later before we do a scatter or whatever.  Under complex step we treat the real and imaginary parts separately as real, then combine them into a complex array when we assign them to the local data vector.  When I updated get_norm I only assigned the real part because it didn't seem right to assign a complex array to the petsc array attribute when we created it using a real array.  Technically though, I guess we don't really need to reset the petsc array at all in get_norm, because it will get set to the proper array later prior to any distriuted operations.

Reply:
ok, i forgot that we did things differently in the petsc array.